### ISSUE TITLE
ezviz  CTQ6C

### ISSUE THREAD

**Issue by maverik80** [RenierM26/ha-ezviz#45]
hi, with the CTQ6C cam I am having problems with the services, they are created but they do not control the cam in its movements and all the rest and in the HA logs I do not see any error inherent to the integration

**Comment by RenierM26** [comment]
Hi @maverik80,

The strings are CASESENSITIVE. Are you typing typing them in uppercase exactly like the example?

**Comment by maverik80** [comment]
> Hi @maverik80,
> 
> The strings are CASESENSITIVE. Are you typing typing them in uppercase exactly like the example?

yes, I see the streaming and I apply the official card but the up down lett and tight services do not respond and from tools if I call the service manually nothing happens but in any case I do not receive err

**Comment by xabaras75** [comment]
> Hi @maverik80,
> 
> The strings are CASESENSITIVE. Are you typing typing them in uppercase exactly like the example?

I have same camera and same issue:
1) I can see the video (streaming works)
2) service not working: no way to use the ptz service. I receive this error message when I push on the UP arrow for example: "Fallita chiamata a servizio ezviz_cloud/ezviz_ptz . 400 Client Error: Bad Request for url: https://apiieu.ezvizlife.com/v3/devices/D61103310/ptzControl"

I have the below card in lovelace (just a part of it):

title: Salone
camera_image: camera.ctq6c_d61103310_salone
camera_view: live
elements:
  - icon: 'mdi:arrow-up'
    style:
      background: 'rgba(255, 255, 255, 0.5)'
      bottom: 50px
      right: 25px
    tap_action:
      action: call-service
      service: ezviz_cloud.ezviz_ptz
      service_data:
        direction: up
        entity_id: camera.ctq6c_d61103310_salone
        speed: 9
    type: icon
  - icon: 'mdi:arrow-down'
    style:
      background: 'rgba(255, 255, 255, 0.5)'
      bottom: 0px
      right: 25px
    tap_action:
      action: call-service
      service: ezviz_cloud.ezviz_ptz
      service_data:
        direction: down
        entity_id: camera.ctq6c_d61103310_salone
        speed: 9
    type: icon
  - icon: 'mdi:arrow-left'
    style:
      background: 'rgba(255, 255, 255, 0.5)'
      bottom: 25px
      right: 50px
    tap_action:
      action: call-service
      service: ezviz_cloud.ezviz_ptz
      service_data:
        direction: left
        entity_id: camera.ctq6c_d61103310_salone
        speed: 9
    type: icon

You mentioned the "The strings are CASESENSITIVE. Are you typing typing them in uppercase exactly like the example?". but in the original card (in the home assistant page, I mean) the direction are written in lowercase. so, are you speaking about a different card?

**Comment by maverik80** [comment]
> > Ciao @ maverik80 ,
> > Le stringhe sono CASESENSITIVE. Stai digitando digitandoli in maiuscolo esattamente come nell'esempio?
> 
> Ho la stessa fotocamera e lo stesso problema:
> 
> 1. Riesco a vedere il video (lo streaming funziona)
> 2. servizio non funzionante: nessun modo per utilizzare il servizio ptz. Ricevo questo messaggio di errore quando premo la freccia SU, ad esempio: "Fallita chiamata a servizio ezviz_cloud / ezviz_ptz. 400 Errore client: richiesta non valida per l'URL: https://apiieu.ezvizlife.com/v3/devices/D61103310/ptzControl "
> 
> Ho la seguente carta in lovelace (solo una parte di essa):
> 
> title: Salone
> camera_image: camera.ctq6c_d61103310_salone
> camera_view: live
> elements:
> 
> * icona: 'mdi: arrow-up' 
>   style: 
>   background: 'rgba (255, 255, 255, 0.5)' 
>   bottom: 50px 
>   right: 
>   25px tap_action: 
>   action: call-service 
>   service: ezviz_cloud.ezviz_ptz 
>   service_data: 
>   direction: up 
>   entity_id: camera.ctq6c_d61103310_salone 
>   velocità: 9 
>   tipo: icona
> * icona: 'mdi: arrow-down' 
>   style: 
>   background: 'rgba (255, 255, 255, 0.5)' 
>   bottom: 0px 
>   right: 
>   25px tap_action: 
>   action: call-service 
>   service: ezviz_cloud.ezviz_ptz 
>   service_data: 
>   direction: down 
>   entity_id: camera.ctq6c_d61103310_salone 
>   velocità: 9 
>   tipo: icona
> * icona: 'mdi: freccia-sinistra' 
>   stile: 
>   sfondo: 'rgba (255, 255, 255, 0.5)' in 
>   basso: 25px 
>   destra: 50px 
>   tap_action: 
>   azione: servizio di chiamata 
>   : ezviz_cloud.ezviz_ptz 
>   service_data: 
>   direzione: sinistra 
>   entity_id: camera.ctq6c_d61103310_salone 
>   velocità: 9 
>   tipo: icona
> 
> Hai menzionato "Le stringhe sono CASESENSITIVE. Stai digitandole in maiuscolo esattamente come nell'esempio?". ma nella scheda originale (nella pagina dell'assistente principale, intendo) le direzioni sono scritte in minuscolo. quindi, stai parlando di una carta diversa?

Unlike I do not receive any type of error but the cam is not commanded, and unfortunately I do not receive errors ... At least I could understand what was wrong

**Comment by nguyenbabao** [comment]
@xabaras75 you can change speed to 5

**Comment by nguyenbabao** [comment]
> hi, with the CTQ6C cam I am having problems with the services, they are created but they do not control the cam in its movements and all the rest and in the HA logs I do not see any error inherent to the integration

your mean is ?
![image](https://user-images.githubusercontent.com/68863522/112110474-83cdb400-8be5-11eb-9b30-e481f1d8455d.png)

**Comment by maverik80** [comment]
> > salve, con la cam CTQ6C ho problemi con i servizi, si creano ma non controllano la cam nei suoi movimenti e tutto il resto e nei log HA non vedo nessun errore inerente all'integrazione
> 
> la tua media è?
> ![Immagine](https://user-images.githubusercontent.com/68863522/112110474-83cdb400-8be5-11eb-9b30-e481f1d8455d.png)
> ![Immagine](https://user-images.githubusercontent.com/68863522/112110612-bf687e00-8be5-11eb-8dd9-2c2fe245f409.png)

![Screenshot_2021-03-23-08-46-24-073_io homeassistant companion android](https://user-images.githubusercontent.com/62015384/112111301-4d2b7580-8bb4-11eb-87a8-a06a08c8cfa7.jpg)

**Comment by RenierM26** [comment]
Hi @maverik80,

Could you please try "speed" with all lowercase characters?

**Comment by maverik80** [comment]
luckily I solved the services they work perfectly but now I have another problem, after the last update 2021.4.0 and firmware update from the ezviz app no ​​longer sends the image to live but only static

**Comment by RenierM26** [comment]
Hi @maverik80,

You could try setting your card to "live" instead of "auto".

**Comment by maverik80** [comment]
> Hi @maverik80,
> 
> You could try setting your card to "live" instead of "auto".

my card is already live but since the HA and camera firmware has been updated the rtsp stream no longer goes

**Comment by RenierM26** [comment]
Hi @maverik80,

Does it show still image or "static" (as in a "snowy" image)? You could also try the latest version that I posted a few minutes back. It contains a fix for some cameras that report a 0.0.0.0 ip. (Which could be the case after the firmware update)

**Comment by maverik80** [comment]
> Ciao @ maverik80 ,
> 
> Mostra un'immagine fissa o "statica" (come in un'immagine "nevosa")? Potresti anche provare l'ultima versione che ho postato qualche minuto fa. Contiene una correzione per alcune telecamere che segnalano un ip 0.0.0.0. (Quale potrebbe essere il caso dopo l'aggiornamento del firmware)

I tried the latest version but nothing ... The new camera firmware seems to block the rtsp stream ...

**Comment by RenierM26** [comment]
@maverik80 ,

Could you try to add /Streaming/channels/101 under the integration options for this camera? It might require a channel for the stream to work. You could also test the stream via VLC media player.

It should result in the following full url:
rtsp://admin: device verification code@IP address :554/Streaming/channels/101

or you could try this as well:

rtsp://admin: device verification code@IP address :554/H.264

**Comment by maverik80** [comment]
> @maverik80 ,
> 
> Could you try to add /Streaming/channels/101 under the integration options for this camera? It might require a channel for the stream to work. You could also test the stream via VLC media player.
> 
> It should result in the following full url:
> rtsp://admin: device verification code@IP address :554/Streaming/channels/101
> 
> or you could try this as well:
> 
> rtsp://admin: device verification code@IP address :554/H.264

In vlc flow rtsp Is ok but in HA no

**Comment by RenierM26** [comment]
Hi @maverik80,

You can add "/H.264" to the camera rtsp stream via the integration options for the camera. (under ffmpeg arguments)
